libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/maneuver/ManeuverView.java

```diff
@@ -118,13 +121,21 @@ public void setManeuverTypeAndModifier(@NonNull String maneuverType, @Nullable S
    * This value will only be considered if the current maneuver type is
    * a roundabout.
    *
-   * @param roundaboutAngle angle to be rendered
+   * @param roundaboutAngle angle to be rendered. If <b>null</b> value is set will be used general
+   *                        roundabout resource.
    */
-  public void setRoundaboutAngle(@FloatRange(from = 60f, to = 300f) float roundaboutAngle) {
-    if (ROUNDABOUT_MANEUVER_TYPES.contains(maneuverType) && this.roundaboutAngle != roundaboutAngle) {
+  public void setRoundaboutAngle(@Nullable @FloatRange(from = 60f, to = 300f) Float roundaboutAngle) {
+    if (ROUNDABOUT_MANEUVER_TYPES.contains(maneuverType) && roundaboutAngle == null) {
```

Comment:
Should we also check for equality here to avoid unnecessary updates?

Reply:
good catch 👍

Reply:
a bit refactored https://github.com/mapbox/mapbox-navigation-android/pull/3867/commits/e1401fe77910d2c05d9193fd8276143d4706c35d

Reply:
LGTM